File: go/problems/simple_is_subsequence.go
package problems

//https://leetcode.com/problems/is-subsequence/
func IsSubsequence(s string, t string) bool {
	tIdx := 0
	ctr := 0
	for _, char := range t {
		if s[tIdx] == uint8(char) {
			ctr += 1
			tIdx += 1
		}
		if len(s) == ctr {
			return true
		}
	}
	return false
}

File: rust/src/problems/simple_is_subsequence.rs
// https://leetcode.com/problems/is-subsequence/submissions/
pub fn is_subsequence(s: String, t: String) -> bool {
    let mut ctr = 0;
    let mut idx = 0;
    for c in t.chars() {
        if s.as_bytes()[idx] == c as u8 {
            ctr += 1;
            idx += 1;
        }
        if ctr == s.len() {
            return true;
        }
    }
    return false;
}